### ISSUE TITLE
consider new theme and GHA for versioned docs

### ISSUE THREAD

**Issue by jGaboardi** [pysal/spaghetti#738]
Consider implementing the [PyData Theme](https://pydata-sphinx-theme.readthedocs.io/en/stable/) from `sphinx`.

* [`pydata-sphinx-theme`](https://github.com/pydata/pydata-sphinx-theme)

Also check out [PyAnsys](https://docs.pyansys.com/)
* https://actions.docs.ansys.com/version/stable/doc-actions/index.html

**Comment by martinfleis** [comment]
I suggest moving this under pysal/pysal and have a discussion on switching the theme across the packages. And there may be better options than pydata-sphinx-theme, like book (used in [xvec](https://xvec.readthedocs.io/en/stable/)) or furo (used by [xyzservices](https://xyzservices.readthedocs.io/en/stable/)). I'd be eventually willing to switch momepy to the same one :).

**Comment by jGaboardi** [comment]
I am very +1 on both suggestions. Just put the issue here so I wouldn't forget.

Think we should migrate to an Issue or Discussion over in `pysal/pysal`?

**Comment by martinfleis** [comment]
Yeah. Since it is actionable and can be closed by a PR, an issue?
